dask/config.py

```diff
@@ -13,19 +14,22 @@
 no_default = "__no_default__"
 
 
-paths = [
+_paths = [
     os.getenv("DASK_ROOT_CONFIG", "/etc/dask"),
     os.path.join(sys.prefix, "etc", "dask"),
+    *[os.path.join(prefix, "etc", "dask") for prefix in site.PREFIXES],
     os.path.join(os.path.expanduser("~"), ".config", "dask"),
     os.path.join(os.path.expanduser("~"), ".dask"),
 ]
 
 if "DASK_CONFIG" in os.environ:
     PATH = os.environ["DASK_CONFIG"]
-    paths.append(PATH)
+    _paths.append(PATH)
 else:
     PATH = os.path.join(os.path.expanduser("~"), ".config", "dask")
 
+# Remove duplicate paths while preserving ordering
+paths = list(dict.fromkeys(_paths))
```

Comment:
Relying on dictionary insertion being ordered is fine. That said this line doesn't preserve the ordering we want. Items that occur later in `_paths` have a higher priority in Dask's config system (i.e. `_paths[0]` is lowest priority while `_paths[-1]` is the highest priority path). 

So I believe we need to add a couple of `reversed(...)` calls

```python
paths = list(reversed(dict.fromkeys(reversed(_paths))))
``` 

to keep the expected ordering. 

Additionally, do you know if `dict.fromkeys` is guaranteed to produce a dictionary with keys who are inserted with the same ordering as the input iterable? Playing around locally, this appears to be the case, but I don't see any referencing to ordering in https://docs.python.org/3/library/stdtypes.html#dict.fromkeys and am wondering if this is just an implementation detail. If `dict.fromkeys` makes this guarantee, then great, let's go with that. Otherwise, I'd prefer to stick with something more explicit like https://github.com/dask/dask/pull/8531#issuecomment-1011379379

Reply:
Whoops missed this comment. Sorry about that: you're absolutely right about me getting the merge order backwards.

My understanding is that `dict.fromkeys` preserves the order of the input iterable. The merging method done here is one that's generally suggested as a good pattern (e.g. see [here](https://blog.finxter.com/how-to-remove-duplicates-from-a-python-list-while-preserving-order/) or [here](https://www.askpython.com/python/remove-duplicate-elements-from-list-python)).

Sending an update with the reversed patch, happy to put in the explicit version if you would still prefer that, just let me know.